### ISSUE TITLE
Use case-insensitive ByteString in EmailAddress

### ISSUE THREAD

**Issue by creichert** [Porges/email-validate-hs#36]
I think it would be appropriate to use the `case-insensitive` package to represent the internal `ByteString`s domain in [`EmailAddress`](https://github.com/Porges/email-validate-hs/blob/master/src/Text/Email/Parser.hs#L23). This would also seem to "fix" the `Eq` instance. 

Has this been considered before? I've generally implemented the entire email address in a web app as `newtype Email = Email CaseInsensitive.ByteString` but after a little research, it appears the part before the `@` can _technically_ be interpreted as case-sensitive.

**Comment by Porges** [comment]
Yes, at the moment there's no normalization performed except for stripping non-significant whitespace and comments. Really the domain part should be handled by a package specifically for handling domains, we could also argue that punycode/Unicode encodings of the same name should be equivalent and that an IPv4-in-IPv6 address should be considered the same as the equivalent address in IPv4. I've started handling the domains more correctly in the work for v3, I'll add this as something to ensure for that work.

You're correct that for the local-part we can't really do anything because the meaning is entirely up to the receiving system.

**Comment by ddssff** [comment]
I don't quite understand how a ByteString can be case insensitive.
